pkg/util/util_test.go

```diff
@@ -163,3 +165,33 @@ func parseIPs(t *testing.T, ipStrs []string) []net.IP {
 
 	return ips
 }
+
+func Test_DeDuplicate(t *testing.T) {
+	tests := map[string]struct {
+		slice    []string
+		expSlice []string
+	}{
+		"nil slice should return nil slice": {
+			slice:    nil,
+			expSlice: nil,
+		},
+		"empty slice should return empty slice": {
+			slice:    []string{},
+			expSlice: []string{},
+		},
+		"slice with no duplicates should return the same slice": {
+			slice:    []string{"a", "b", "c"},
+			expSlice: []string{"a", "b", "c"},
+		},
+		"slice with duplicates should return the same slice with the duplicates removed": {
+			slice:    []string{"a", "b", "c", "a", "b", "c", "d", "e", "f", "a"},
+			expSlice: []string{"a", "b", "c", "d", "e", "f"},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			assert.Equal(t, test.expSlice, DeDuplicate(test.slice), "%v", test.slice)
+		})
+	}
+}
```

Comment:
I wondered if you could use `sets.NewString` instead of `DeDuplicate`? 
Is it important to preserve the order of the original slice?
If so, perhaps there should be a comment in the code about that and a unit-test that demonstrates the order of slice is maintained....in other words a test that would fail if I replace `DeDuplicate` here with `sets.NewString(test.slice).List()` here.

Reply:
Thanks. `sets.String` seems a lot cleaner to me, so have replaced the slices in the SecretTempalte controller in favor of that.

I've also removed `DeDuplicate` and its tests since it is no longer used.